Assets/Mirror/Runtime/NetworkServer.cs

```diff
@@ -77,6 +77,8 @@ public static class NetworkServer
         /// </summary>
         static readonly List<int> connectionIdsCache = new List<int>();
 
+        static readonly List<NetworkIdentity> dirtyObjects = new List<NetworkIdentity>();
```

Comment:
This should be a set instead of list.

why?

Suppose you add 10 elements to a list,  onDirty will be raised 10 times, and the same will be added here 10 times so it will be synchronized 10 times.